### DISCUSSION TITLE
Motif supports Markdoc

### DISCUSSION THREAD

**Discussion by michaelfester** [markdoc/markdoc#43] [Show and tell]
You can now author Markdoc pages straight from the browser using [Motif](https://motif.land). Just add the `.mdoc` extension to pages that should be parsed as Markdoc.

## Configuration

To add a custom Markdoc [configuration](https://markdoc.io/docs/syntax#config), create a file named `markdoc.config.js` at the root of the project source. It should adhere to the ES module syntax, and should export a default value holding the config object. Here is an example:

```js
// markdoc.config.js
const config = {
  nodes: { heading: { render: 'Heading', children: ['paragraph'] } },
  tags: { popover: { render: 'Popover' } },
  variables: { name: 'Dr. Mark' },
};

export default config
```

## Components

To enable custom components, as referenced from the config's [`nodes`](https://markdoc.io/docs/nodes) and [`tags`](https://markdoc.io/docs/tags) entries, create a file named `markdoc.components.js` at the root of the project source. It should also adhere to the ES module syntax, and should export a default value holding the components. It supports ES modules imports, both from the web and from other parts of the project:

```js
// markdoc.components.js
import { Popover } from 'https://cdn.skypack.dev/@headlessui/react@1.4.3'
import { HeadingWrapper } from "@components/wrappers"

const Heading = ({ id, children }) => {
  return <HeadingWrapper id={id}>{ children }</HeadingWrapper>
}

const components = {
  Popover,
  Heading
}

export default components
```

## Further reading

Full docs here: https://motif.land/docs/basics/markdoc.

Feedback welcome!

**Comment by mfix-stripe** [comment]
This is incredible @michaelfester! Thanks for jumping on it so fast.
